api/types/src/transaction.rs

```diff
@@ -586,13 +579,197 @@ pub struct BlockMetadataTransaction {
+            ValidatorTransaction::DkgResult(t) => &t.info,
+        }
+    }
+
+    pub fn transaction_info_mut(&mut self) -> &mut TransactionInfo {
+        match self {
+            ValidatorTransaction::ObservedJwkUpdate(t) => &mut t.info,
+            ValidatorTransaction::DkgResult(t) => &mut t.info,
+        }
+    }
+
+    pub fn timestamp(&self) -> U64 {
+        match self {
+            ValidatorTransaction::ObservedJwkUpdate(t) => t.timestamp,
+            ValidatorTransaction::DkgResult(t) => t.timestamp,
+        }
+    }
+
+    pub fn from_internal_repr(
+        internal: aptos_types::validator_txn::ValidatorTransaction,
+        info: TransactionInfo,
+        events: Vec<Event>,
+        timestamp: u64,
+    ) -> Self {
+        match internal {
+            aptos_types::validator_txn::ValidatorTransaction::DKGResult(dkg_transcript) => {
+                Self::DkgResult(DKGResultTransaction {
+                    info,
+                    events,
+                    timestamp: U64::from(timestamp),
+                    dkg_transcript: ExportedDKGTranscript::from_internal_repr(dkg_transcript),
+                })
+            },
+            aptos_types::validator_txn::ValidatorTransaction::ObservedJWKUpdate(
+                quorum_certified_update,
+            ) => Self::ObservedJwkUpdate(JWKUpdateTransaction {
+                info,
+                events,
+                timestamp: U64::from(timestamp),
+                quorum_certified_update: ExportedQuorumCertifiedUpdate::from_internal_repr(
+                    quorum_certified_update,
+                ),
+            }),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct JWKUpdateTransaction {
+    #[serde(flatten)]
+    #[oai(flatten)]
+    pub info: TransactionInfo,
+    pub events: Vec<Event>,
+    pub timestamp: U64,
+    pub quorum_certified_update: ExportedQuorumCertifiedUpdate,
+}
+
+/// A more API-friendly representation of the on-chain `aptos_types::jwks::QuorumCertifiedUpdate`.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct ExportedQuorumCertifiedUpdate {
+    pub update: ExportedProviderJWKs,
+    pub multi_sig: ExportedAggregateSignature,
+}
+
+impl ExportedQuorumCertifiedUpdate {
+    pub fn from_internal_repr(internal: QuorumCertifiedUpdate) -> Self {
+        let QuorumCertifiedUpdate { update, multi_sig } = internal;
+        Self {
+            update: ExportedProviderJWKs::from_internal_repr(update),
+            multi_sig: ExportedAggregateSignature::from_internal_repr(multi_sig),
+        }
+    }
+}
+
+/// A more API-friendly representation of the on-chain `aptos_types::aggregate_signature::AggregateSignature`.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct ExportedAggregateSignature {
+    signer_indices: Vec<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sig: Option<Vec<u8>>,
```

Comment:
Also, I all Vec<U8> I think should be using SerdeBytes?  Unsure how this comes out

Reply:
you mean `#[serde(with = "serde_bytes")]` right? it doesn't seem to make a difference. With/without it, the serialized value is like `[0, 1, 2, ...]`

Reply:
i think the existing `HexEncodedBytes` helps. updated to that.